core/src/test/java/hudson/PluginWrapperTest.java

```diff
@@ -95,6 +100,35 @@ public void dependencyFailedToLoad() {
         assertContains(ex, "Failed to load: dependee (42)", "Failed to load: dependency (5)");
     }
 
+    @Issue("JENKINS-66563")
+    @Test
+    public void insertJarsIntoClassPath() throws Exception {
+        try (AntClassLoader cl = new AntClassLoader()) {
+            assertInjectingJarsWorks(cl);
+        }
+    }
+
+    @Issue("JENKINS-66563")
+    @Test
+    public void insertJarsIntoClassPathURLCL() throws Exception {
+        try (URLClassLoader2 cl = new URLClassLoader2(new URL[0])) {
+            assertInjectingJarsWorks(cl);
+        }
+    }
+
+    private void assertInjectingJarsWorks(ClassLoader cl) throws Exception {
+        PluginWrapper pw = pluginWrapper("pw").version("1").classloader(cl).build();
+        Enumeration<?> e1 = pw.classLoader.getResources("META-INF/MANIFEST.MF");
```

Comment:
@basil FYI: one interesting thing I found writing this test is that the Enumeration for URL2Classloader is lazy, whereas it is not for ANTCLassLoader.
in other words if you do not count the size of `e1` before adding the new jar but delay it you end up counting the same size.  Not sure if that will cause any subtle issues in Jenkins, but something to potentially keep in mind.